lib/Common/Memory/RecyclerTelemetryInfo.cpp

```diff
@@ -0,0 +1,256 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include "CommonMemoryPch.h"
+
+
+#ifdef ENABLE_BASIC_TELEMETRY
+
+#include "Recycler.h"
+#include "HeapBucketStats.h"
+#include "BucketStatsReporter.h"
+#include <psapi.h>
+
+namespace Memory
+{
+#ifdef DBG
+#define AssertOnMainThread(recyclerTelemetryInfo, loc) { AssertMsg(recyclerTelemetryInfo->mainThreadID == GetCurrentThreadContextId(), STRINGIZE("Unexpected thread ID at " ## loc)); }
+#else
+#define AssertOnMainThread(recyclerTelemetryInfo, loc)
+#endif
+
+    RecyclerTelemetryInfo::RecyclerTelemetryInfo(Recycler * recycler, RecyclerTelemetryHostInterface* hostInterface) :
+        passCount(0),
+        hostInterface(hostInterface),
+        lastPassStats(nullptr),
+        recyclerStartTime(Js::Tick::Now()),
+        captureTelemetryData(true)
+    {
+        this->recycler = recycler;
+
+        // temp
+        mainThreadID = GetCurrentThreadContextId();
+    }
+
+    const GUID& RecyclerTelemetryInfo::GetRecyclerID() const
+    {
+        return this->recycler->GetRecyclerID();
+    }
+
+    bool RecyclerTelemetryInfo::GetIsConcurrentEnabled() const
+    { 
+        return this->recycler->IsConcurrentEnabled(); 
+    }
+
+    RecyclerTelemetryInfo::~RecyclerTelemetryInfo()
+    {
+        AssertOnMainThread(this, RecyclerTelemetryInfo::~RecyclerTelemetryInfo);
+        this->FreeGCPassStats();
+    }
+
+    void RecyclerTelemetryInfo::FillInSizeData(IdleDecommitPageAllocator* allocator, AllocatorSizes* sizes) const
+    {
+        sizes->committedBytes = allocator->GetCommittedBytes();
+        sizes->reservedBytes = allocator->GetReservedBytes();
+        sizes->usedBytes = allocator->GetUsedBytes();
+        sizes->numberOfSegments = allocator->GetNumberOfSegments();
+    }
+
+    void RecyclerTelemetryInfo::StartPass()
+    {
+        Js::Tick start = Js::Tick::Now();
+        if(!this->captureTelemetryData)
+        { 
+            return;
+        }
+
+        //Output::Print(L"startPass called (current thread = %p, main thread = %p), passCount  = %i\r\n", GetCurrentThreadContextId(), this->mainThreadID, this->passCount);
```

Comment:
Remove commented code? Alternatively, add a Phase to ConfigFlagsList.h and use the PHASE_TRACE macros

Reply:
removed